packages/protocol/contracts/L1/libs/LibProving.sol

```diff
@@ -459,4 +451,24 @@ library LibProving {
             header.mixHash != meta.mixHash
         ) revert L1_META_MISMATCH();
     }
+
+    function _getInstance(
+        Evidence memory evidence
+    ) internal pure returns (bytes32 instance) {
```

Comment:
@smtmfft @davidtaikocha I'm wonder, if the circuits really need all inputs to be RLP encoded first, won't packing different bytes32 together be more efficient?

Reply:
I think there are two reasons:
1. RLP encoding can be used for more complex data structures, such as list, nested list, etc. (Assuming pi will change in the future
2. The main reason ABI encoding and RLP encoding will require two different circuits, but they actually do the same thing. RLP is more versatile than ABI with a little overhead. More circuits make optimization difficult.